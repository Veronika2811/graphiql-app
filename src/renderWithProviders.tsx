import { PropsWithChildren } from 'react';
import { Provider } from 'react-redux';
import { HashRouter } from 'react-router-dom';
import { render, type RenderOptions } from '@testing-library/react';
import { SnackbarProvider } from 'context/snackbar-provider';
import { LocaleProvider } from 'internationalization/LocaleProvider';
import { MuiProvider } from 'layouts/mui';
import { AppStore, setupStore } from 'store';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  store?: AppStore;
}

export const renderWithProviders = (
  ui: React.ReactElement,
  { store = setupStore(), ...renderOptions }: ExtendedRenderOptions = {}
) => {
  const Wrapper = ({ children }: PropsWithChildren): JSX.Element => (
    <MuiProvider>
      <SnackbarProvider>
        <HashRouter>
          <Provider store={store}>
            <LocaleProvider>{children}</LocaleProvider>
          </Provider>
        </HashRouter>
      </SnackbarProvider>
    </MuiProvider>
  );
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};

export default renderWithProviders;
