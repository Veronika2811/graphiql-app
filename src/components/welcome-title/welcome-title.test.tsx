import { renderWithProviders } from '../../renderWithProviders';

import { WelcomeTitle } from '.';

describe('WelcomeTitle component', () => {
  it('should renders correctly WelcomeTitle component', () => {
    const container = renderWithProviders(<WelcomeTitle />);
    expect(container).toMatchSnapshot();
  });
});
