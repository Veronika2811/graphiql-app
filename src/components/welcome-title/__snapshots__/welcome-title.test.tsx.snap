// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WelcomeTitle component > should renders correctly WelcomeTitle component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-yd8sa2"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 MuiTypography-alignCenter css-4d4fmw-MuiTypography-root"
        >
          The Best IDE for GraphQL Requests
        </h1>
        <p
          class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-alignCenter css-1ukqcrx-MuiTypography-root"
        >
          The app is the result of completing the final task of RS School React Course.
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-yd8sa2"
    >
      <h1
        class="MuiTypography-root MuiTypography-h1 MuiTypography-alignCenter css-4d4fmw-MuiTypography-root"
      >
        The Best IDE for GraphQL Requests
      </h1>
      <p
        class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-alignCenter css-1ukqcrx-MuiTypography-root"
      >
        The app is the result of completing the final task of RS School React Course.
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
